src/main/java/net/smartcosmos/dao/metadata/MetadataDao.java

```diff
@@ -14,55 +15,84 @@
 public interface MetadataDao {
 
     /**
-     * Upserts a list of metadata entities associated to a reference entity in the realm of a given account.
+     * Inserts a list of metadata entities associated to a reference entity in the realm of a given tenant.
      *
-     * @param accountUrn the account URN
-     * @param upsertMetadataCollection the collection containing the metadata entries to upsert
-     * @return an {@link MetadataResponse} list for the upserted metadata
-     * @throws ConstraintViolationException if the {@link MetadataUpsert} violates constraints enforced by the persistence service
+     * @param tenantId the tenant URN
+     * @param createMetadataCollection the collection containing the metadata entries to insert
+     * @return an {@link MetadataResponse} list for the inserted metadata
+     * @throws ConstraintViolationException if the {@link MetadataCreate} violates constraints enforced by the persistence service
      */
-    List<MetadataResponse> upsert(String accountUrn, Collection<MetadataUpsert> upsertMetadataCollection) throws ConstraintViolationException;
+    List<MetadataResponse> create(String tenantId, Collection<MetadataCreate> createMetadataCollection)
```

Comment:
The signature of this method should either be `create(String tenantId, String ownerType, String ownerId, Map<String, Object> metadata)` or `create(String tenantId, MetadataCreate createMetadata)`.

You can't create metadata for multiple things at once, but you can create multiple metadata entries at once. Therefore, the key-value `Map<String, Object>` needs to be supported somewhere, but creating full-blown `CreateMetadata` instances for each and every key-value pair is too much overhead.